lib/ibanity/api/base_resource.rb

```diff
@@ -97,22 +97,31 @@ def prepare_attributes(raw)
 
     def setup_relationships(relationships, customer_access_token = nil)
       relationships.each do |key, relationship|
-        if relationship["data"]
-          self[Ibanity::Util.underscore("#{key}_id")] = relationship["data"]["id"]
-          return unless relationship.dig("links", "related")
-
-          klass = relationship_klass(key)
-          method_name = Ibanity::Util.underscore(key)
-          define_singleton_method(method_name) do |headers: nil|
-            klass.find_by_uri(uri: relationship["links"]["related"], headers: headers, customer_access_token: customer_access_token)
-          end
-        else
-          singular_key = key[0..-2]
-          klass        = relationship_klass(singular_key)
-          method_name  = Ibanity::Util.underscore(key)
-          define_singleton_method(method_name) do |headers: nil, **query_params|
-            uri = relationship["links"]["related"]
-            klass.list_by_uri(uri: uri, headers: headers, query_params: query_params, customer_access_token: customer_access_token)
+        url = relationship.dig("links", "related")
+        if url
+          if relationship["data"]
+            resource = relationship.dig("data", "type") || key
+            klass = relationship_klass(resource)
+            method_name = Ibanity::Util.underscore(resource)
```

Comment:
It makes sense to get the class from the data type, but shouldn't we continue to use the relationship key for the method name and id attribute? For backward compatibility (of the id at least, I suppose the method was broken) and in case there are multiple relationships with the same data type.

Reply:
indeed the method name should still be the one of the relationship. Getting the class from the type is correct